zerver/lib/users.py

```diff
@@ -174,28 +176,24 @@ def is_administrator_role(role: int) -> bool:
 
 def bulk_get_cross_realm_bots() -> Dict[str, UserProfile]:
```

Comment:
Hmm, I think this direction might not be helpful because we are mid-migration towards eliminating cross-realm bots in favor of just creating those bots in each realm? Not sure, I gotta run but I'm worried that this set of changes might complicate adding a `realm_id` parameter to find these.

Reply:
I don't really share your concern here.  This removes a bunch of complicated code related to the current paradigm.  Once the system bots belong to the same realm, we simply tweak the query instead of mucking around with cache_key_function, query_function, object_ids, and id_fetcher

Reply:
Insofar as we would need to add a `realm_id` parameter, it's worth noting that this commit doesn't change any function signatures.  So the work to plumb that it would be the same regardless of whether you add this; it's just a a question of how long you want some future dev staring at all the complicated cache-related code.

Reply:
See also https://chat.zulip.org/#narrow/stream/3-backend/topic/bulk_get_streams.20slowness/near/1613575